### ISSUE TITLE
Why do self defined colors behave different

### ISSUE THREAD

**Issue by SieDa** [plantuml/plantuml#1948]
As outlined in chapter 5 Colors of the hitchhikers guide, it provides a procedure to draw a color box. If I
`!define mycolorBlue 4A80B6`
and use the procedure to output
`$DrawColor("mycolorBlue") `
the result is not a box with my color-name inside but a box named with 4A80B6.
So it behaves different then the predefined colors. Why and how to change that? 

```plantuml
@startuml

!unquoted procedure $DrawColor($colour)

    skinparam rectangle {
            backgroundColor<<$colour>> $colour
            borderColor<<$colour>> $colour
            shadowing<<$colour>> true
            BorderThickness<<$colour>> 1
        }

    rectangle $colour <<$colour>> as "<color:$colour></color>"

!endprocedure

!define mycolorBlue 4A80B6
package mycolors {
$DrawColor("mycolorBlue")
}
@enduml
```
![376324722-03c1f02b-f50e-40fc-890d-133813ef79ed](https://github.com/user-attachments/assets/73fd9cbf-83e3-4bd6-82b1-93d6e50043f9)

Can you provide a simple example for color definitions and their documentation, can you explain why it does not work how the guide suggests it and how to change it?

1. How to fix the procedure to provide my color name "mycolorBlue" in the box?
2. How to povide a second color to the procedure for the color name to be coloured by mycolorLightBlue defined by F8F7FF?
3. On top to that effect, it is not possible to define relations between such entities, defined by that procedure, like:

`mycolorBlue --> mycolorGreen`

**Comment by The-Lum** [comment]
Hello @SieDa,

For that you can use JSON structure and 'hide the stereotype' as:
```puml
@startuml

!$mycolor={
  "mycolorBlue": "4A80B6",
  "mycolorLightBlue": "F8F7FF",
  "mycolorGreen": "C0FFC0",
  "default": "000000"
}

hide stereotype
!unquoted procedure $DrawColor($colour, $colour_name=default)

    skinparam rectangle {
            backgroundColor<<$colour>> $mycolor[$colour]
            borderColor<<$colour>> $mycolor[$colour]
            shadowing<<$colour>> true
            BorderThickness<<$colour>> 1
        }

    rectangle "<color:$mycolor[$colour_name]>$colour</color>" as $colour <<$colour>>

!endprocedure


package mycolors {
  $DrawColor(mycolorBlue, mycolorLightBlue)
}
$DrawColor(mycolorGreen)

mycolorBlue -> mycolorGreen
@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/bL9DJyCm3BtdLtZH3amQeaW8qDHLOqFbmf4Rcb1eJ5kjJKe-D4tJ_ZjfbevidF0foREVu-Utq8OfOzk6aD6utHMoaMg-9m0qN9QDHJe3UlVqc2plwJI2NkkoCWEUF-OFUNw6loX4qMEh9CzNoH7Z-CLiO_hdn0Sb1qAgcYDeWmgbsNL8HbPyMscGGwTaWTmgXF6pOjjLJtmrxlcjca8uV0ZMuZnGJuZh0w0tjUYOOYqeB0mJPOEmzyWGdwpOb4fQmJrhcWQsB8D1YlVmj3ulb8gZ-cUHhXYNsrgKSObH5i-obfxwhQgBZK2juzpRK-BXEE7lN3JrZMUNF_2whBDmIsyyc55WUb0EeWxE0YZuIN12IESKOYL2eDLUm6WFaKUcSEcAYLlitrnl2hUYg1IkCuXHid3VS9Ry0G00)](https://www.planttext.com/?text=bL9DJyCm3BtdLtZH3amQeaW8qDHLOqFbmf4Rcb1eJ5kjJKe-D4tJ_ZjfbevidF0foREVu-Utq8OfOzk6aD6utHMoaMg-9m0qN9QDHJe3UlVqc2plwJI2NkkoCWEUF-OFUNw6loX4qMEh9CzNoH7Z-CLiO_hdn0Sb1qAgcYDeWmgbsNL8HbPyMscGGwTaWTmgXF6pOjjLJtmrxlcjca8uV0ZMuZnGJuZh0w0tjUYOOYqeB0mJPOEmzyWGdwpOb4fQmJrhcWQsB8D1YlVmj3ulb8gZ-cUHhXYNsrgKSObH5i-obfxwhQgBZK2juzpRK-BXEE7lN3JrZMUNF_2whBDmIsyyc55WUb0EeWxE0YZuIN12IESKOYL2eDLUm6WFaKUcSEcAYLlitrnl2hUYg1IkCuXHid3VS9Ry0G00) 

_Enjoy,_
Regards,
Th.

**Comment by The-Lum** [comment]
And FYI:
- here is another answer using new `style` instead old `skinparam` as:
```puml
@startuml

!$mycolor={
  "mycolorBlue": "4A80B6",
  "mycolorLightBlue": "F8F7FF",
  "mycolorGreen": "C0FFC0",
  "default": "000000"
}

hide stereotype
!unquoted procedure $DrawColor($colour, $colour_name=default)
  <style>
  .$colour {
    BackgroundColor $mycolor[$colour]
    LineColor $mycolor[$colour]
    Shadowing 1
    LineThickness 1
  }
  </style>
  rectangle "<color:$mycolor[$colour_name]>$colour</color>" as $colour <<$colour>>
!endprocedure


package mycolors {
  $DrawColor(mycolorBlue, mycolorLightBlue)
}
$DrawColor(mycolorGreen)

mycolorBlue -> mycolorGreen
@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/TPBVIyCm4CVVxw-uXJqugAw2w1XTcPlKbxtfsnWIchCjQvEP7umX-zzjOofrOfvo-Lo-kUzTbiPoRLrJ0up6pIbNjTABJq1a8LZLZjWSsTtZB5xTisY0DbLHsfvdi-mXotxnPqqaExQEismTVpD1xzpLjZkE_M9m1YWhGMWiQLBsT20OEVdXb2M11wro4auJZfyqFwuxuQjnf-zqX67p9db3Yo0zWSJOKqqft0ICdQ4Lp_U5LauAhu6zssr8shKvcqhI__Ibv48TArdWRSXzBQjyBycOzkGCoJIygoctN1Or8KkyolnIpXUyIqEKJ3rC6NBJMy8a2RiqXH59yTC803YqNdX165IDDpZeps1s4LvEQzAM-ZVN3si2CBYAroaEAIpRCjglyWK0)](https://www.planttext.com/?text=TPBVIyCm4CVVxw-uXJqugAw2w1XTcPlKbxtfsnWIchCjQvEP7umX-zzjOofrOfvo-Lo-kUzTbiPoRLrJ0up6pIbNjTABJq1a8LZLZjWSsTtZB5xTisY0DbLHsfvdi-mXotxnPqqaExQEismTVpD1xzpLjZkE_M9m1YWhGMWiQLBsT20OEVdXb2M11wro4auJZfyqFwuxuQjnf-zqX67p9db3Yo0zWSJOKqqft0ICdQ4Lp_U5LauAhu6zssr8shKvcqhI__Ibv48TArdWRSXzBQjyBycOzkGCoJIygoctN1Or8KkyolnIpXUyIqEKJ3rC6NBJMy8a2RiqXH59yTC803YqNdX165IDDpZeps1s4LvEQzAM-ZVN3si2CBYAroaEAIpRCjglyWK0) 

_Enjoy,_
Th.

**Comment by The-Lum** [comment]
Hi @SieDa 

- Would my answers suit you?

Regards,
Th.

**Comment by SieDa** [comment]
Yes - sorry for the late feedback. I could apply <<styles>> and looking into the source code that is wide spread supported now.
